test/extended/cpu_partitioning/managed.go

```diff
@@ -0,0 +1,189 @@
+package cpu_partitioning
+
+import (
+	"context"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+
+	appv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/image"
+)
+
+var _ = g.Describe("[sig-node] CPU Partitioned cluster workloads", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc  = exutil.NewCLIWithoutNamespace("cpu-partitioning").AsAdmin()
+		ctx = context.Background()
+	)
+
+	g.BeforeEach(func() {
+		skipNonCPUPartitionedCluster(oc)
+	})
+
+	g.Context("in annotated namespaces", func() {
+
+		var (
+			deploymentLabels = map[string]string{"app": "workload-pinned"}
+			namespace        = "workload-pinning"
+			name             = "workload"
+		)
+
+		g.AfterEach(func() {
+			o.Expect(cleanup(oc, namespace)).NotTo(o.HaveOccurred())
+		})
+
+		g.It("should be modified", func() {
+
+			e := createNamespace(oc, namespace, namespaceAnnotation)
+			o.Expect(e).ToNot(o.HaveOccurred(), "error creating namespace %s", namespace)
+
+			e = createDeployment(oc, name, namespace, deploymentLabels, deploymentPodAnnotation)
+			o.Expect(e).ToNot(o.HaveOccurred(), "error creating pinned deployment")
+
+			pods, err := oc.KubeClient().CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+				LabelSelector: "app=workload-pinned",
+			})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			for _, pod := range pods.Items {
+				o.Expect(pod.Annotations).To(
+					o.And(
+						o.HaveKey(workloadAnnotations),
+						o.HaveKey(o.MatchRegexp(workloadAnnotationsRegex)),
+					), "pod (%s/%s) does not contain correct annotations", pod.Namespace, pod.Name)
+
+				for _, container := range pod.Spec.Containers {
+					_, ok := container.Resources.Limits[resourceLabel]
+					o.Expect(ok).To(o.BeTrue())
+					_, ok = container.Resources.Requests[resourceLabel]
+					o.Expect(ok).To(o.BeTrue())
+				}
+			}
+		})
+	})
+
+	g.Context("in non-annotated namespaces", func() {
```

Comment:
shouldn't this test also be done when the feature isn't enabled?

Reply:
actually both of these.  you need to ensure that nothin attempts to use these, right?

Reply:
That's a great idea, I went ahead and updated all the tests to be able to run on both cpu partitioned and non cpu partitioned clusters. This will allow us to catch deployments and namespaces that aren't annotated correctly.

I tested this on both types of clusters and found a couple of deployments that aren't annotated correctly so I have it skipping one of those tests for now on a non-cpu partitioned cluster. 

I'm Thinking we merge this in now, get the pipeline running for CPU partitioned clusters while I'll go open up PRs to update those deployments to fix it running for all clusters. What do you think?